SwiftLeeds/Views/Local/LocalView.swift

```diff
@@ -7,29 +7,66 @@
     @StateObject private var model = LocalViewModel()
 
-    
+    //-1.548567
+    @State private var bottomSheetShown = true
+    @State private var mapRegion: MKCoordinateRegion = MKCoordinateRegion(center: CLLocationCoordinate2D(latitude: 53.78613099154973, longitude: -1.5461652186147719), span: MKCoordinateSpan(latitudeDelta: 0.04, longitudeDelta: 0.04))
+    @State private var selectedLocation: Local.Location?
+
     var body: some View {
-        GeometryReader{ geometry in
-            ZStack {
-                Map(coordinateRegion: $mapRegion)
+        ZStack {
+            GeometryReader{ geometry in
+                if let category = model.selectedCategory {
+                    Map(coordinateRegion: $mapRegion, showsUserLocation: true, annotationItems: category.locations) { location in
+                        MapAnnotation(coordinate: location.location.coordinate) {
+                            Image(uiImage: UIImage(systemName: category.symbolName) ?? UIImage(imageLiteralResourceName: category.symbolName))
+                                .frame(width: 44, height: 44)
+                                .background(
+                                    RoundedRectangle(cornerRadius: 6)
+                                        .fill(.white)
+                                )
+                                .onTapGesture {
+                                    selectedLocation = location
+                                    print(mapRegion)
+                                }
+                        }
+                    }
                     .ignoresSafeArea()
+                } else {
+                    Map(coordinateRegion: $mapRegion, showsUserLocation: true)
+                        .ignoresSafeArea()
+                }
+
+                if let location = selectedLocation {
+                    ZStack {
+                        Color.black.opacity(0.3)
+                            .ignoresSafeArea(.all)
+                            .onTapGesture {
+                                selectedLocation = nil
+                            }
+
+                        locationInfoView(category: model.selectedCategory!, location: location)
+                            .padding(.bottom, bottomSheetShown ? geometry.size.height * Constants.maxHeightRatio: 0)
+                            .animation(.easeInOut, value: bottomSheetShown)
+                    }
+                }
+
                 BottomSheetView(
-                    isOpen: self.$bottomSheetShown,
+                    isOpen: $bottomSheetShown,
+                    selectedCategory: $model.selectedCategory,
                     categories: model.categories,
                     error: model.error,
                     maxHeight: geometry.size.height * Constants.maxHeightRatio
                 )
+
                 if model.error != nil {
                     errorView
                 }
             }
         }
```

Comment:
If we add a cheeky animation modifier here:
```
 .animation(.easeInOut, value: selectedLocation)
```

And one of these lovely things below line 53:
```
.transition(.opacity)
```

We'll get this fading in for free 👯

Reply:
I'm not sure which part this is referring to?